engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/PeriodicSourceFactory.scala

```diff
@@ -0,0 +1,99 @@
+package pl.touk.nussknacker.engine.flink.util.transformer
+
+import java.time.Duration
+import java.{util => jul}
+
+import javax.annotation.Nullable
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.api.scala._
+import org.apache.flink.runtime.state.{FunctionInitializationContext, FunctionSnapshotContext}
+import org.apache.flink.streaming.api.TimeCharacteristic
+import org.apache.flink.streaming.api.checkpoint.CheckpointedFunction
+import org.apache.flink.streaming.api.functions.source.SourceFunction
+import org.apache.flink.streaming.api.functions.timestamps.AscendingTimestampExtractor
+import org.apache.flink.streaming.api.functions.{AssignerWithPeriodicWatermarks, AssignerWithPunctuatedWatermarks, TimestampAssigner}
+import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.process.Source
+import pl.touk.nussknacker.engine.api.typed.{ReturningType, typing}
+import pl.touk.nussknacker.engine.flink.api.process.{FlinkCustomNodeContext, FlinkSource, FlinkSourceFactory}
+
+import scala.collection.JavaConverters._
+
+// TODO: add testing capabilities
+object PeriodicSourceFactory extends PeriodicSourceFactory(new MapAscendingTimestampExtractor(MapAscendingTimestampExtractor.DefaultTimestampField))
+
+class PeriodicSourceFactory(timestampAssigner: TimestampAssigner[jul.Map[String, AnyRef]]) extends FlinkSourceFactory[jul.Map[String, AnyRef]]  {
+
+  @MethodToInvoke
+  def create(@ParamName("period") period: Duration,
+             // TODO: @DefaultValue(1) instead of nullable
+             @ParamName("count") @Nullable nullableCount: Integer,
+             @ParamName("value") value: LazyParameter[jul.Map[String, AnyRef]]): Source[_] = {
```

Comment:
does it have to be map? Or just AnyRef? I know you have MapAscendingTimestampExtractor below, but for constant time field maybe we could have just another field?

Reply:
Good point, changed to AnyRef. Would be nice to have `@Placeholder({:})` which not works now :/